### DISCUSSION TITLE
[Shipped] Support for Node.js 16.x (LTS)

### DISCUSSION THREAD

**Discussion by kaleabmelkie** [vercel/community#37] [Idea]
**Edit:** Node.js 16 LTS support has been released: https://twitter.com/vercel_changes/status/1523730208750252033

---

Currently, it seems that only Node.js 12.x and 14.x are supported.

Is there a temporary workaround (or a timeline) for shipping apps that need the 16.x LTS version on Vercel?

**Comment by leerob** [comment]
Edit: [Node.js 16 LTS support has been released.](https://twitter.com/vercel_changes/status/1523730208750252033)

**Reply by earlybail** [reply]
Hello, is there any update on this?

**Reply by Pieparker** [reply]
Still waiting on https://github.com/aws/aws-lambda-base-images/issues/14 y'all!

**Reply by wokejacqueline** [reply]
@Pieparker any update on this? looks like aws updated the lambda base images to support node 16.  any idea on vercel's timeline? :)

**Reply by Pieparker** [reply]
[Not yet](https://docs.aws.amazon.com/lambda/latest/dg/lambda-nodejs.html)!

**Comment by pcjmfranken** [comment]
If the overhead is worth it to you, you could create a [custom Vercel runtime](https://vercel.com/docs/runtimes#advanced-usage/developing-your-own-runtime) and [have it run](https://github.com/vercel/vercel/blob/main/DEVELOPING_A_RUNTIME.md#lambdaruntime) a [custom AWS Lambda runtime](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-custom.html)[^1] (your own or someone else's[^2]) with whatever Node.js version you want.

Maybe if we ask @leerob really kindly he could get Vercel to provide one[^3]? wink wink

[^1]: Post from an AWS dev with some interesting insight in how the Lambda runtimes are put together: https://github.com/aws/aws-lambda-base-images/issues/7#issuecomment-771083483
[^2]: Random find, run at your own risk: https://github.com/janaz/aws-lambda-node-runtime
[^3]: And silently switch to the official AWS Node 16 lambda runtime when it's out

**Comment by kvetoslavnovak** [comment]
+1
To replace all my usage of the String.prototype.replaceAll() string method in my backend code only because of Vercel Node.js 14.x limit is very frustrating TBH :-P

**Comment by fierysolid** [comment]
Maybe consider also offering Cloud Functions from Google as an option since AWS doesn't seem to care about updating their base images in a timely manner.

**Comment by mattpilott** [comment]
I've just created and then cancelled a pro account due to node 14 still being the latest option on vercel. Will return when 16 is available!

**Comment by bradennapier** [comment]
Documentation should be updated as you say you support the latest LTS but you don't - you support the latest LTS that AWS Lambda natively supports :-P

**Comment by chronos-ohm** [comment]
Consider adding support for other cloud providers. Depending on AWS to update node version support is not a good idea.

**Comment by michaelhays** [comment]
Fusebit recently released [everynode](https://github.com/fusebit/everynode), which:

> allows you to run any version of Node.js in AWS Lambda ... within six hours of the release.

See [this comment](https://github.com/aws/aws-lambda-base-images/issues/14#issuecomment-1040655827) in the Node v16 AWS issue for more context.

Would Vercel be willing to add support for this? It appears to be well-tested. This would be extremely helpful, since AWS appears to take a full calendar year to add support for Node.js versions after their initial release. I'd love to be able to use Node.js v18 in Vercel Functions in April 2022 🙏

**Comment by sedhha** [comment]
I wish Vercel could use GCP... AWS downtimes and issues have been so much pain at many times to me!

**Comment by alfosua** [comment]
AWS dead time is a joke. It seems that they may wait for a full maintenance release:

https://nodejs.org/en/about/releases/

And Node 16 Maintenance LTS release will do about October, this year! That's a lot of time to wait for!

I think @pcjmfranken's proposal would really help for the next 8 months. You could leave a note that the runtime may not be stable.

**Comment by leerob** [comment]
Hey everyone! We're extremely close, looking at a release in the next few weeks: https://github.com/aws/aws-lambda-base-images/issues/14#issuecomment-1108679026

Thank for your being patient here 🙏

**Comment by leerob** [comment]
Great news everyone, Node.js 16 LTS support has been released! 🎉 

https://twitter.com/vercel_changes/status/1523730208750252033
